libs/marshalling/algebra/include/nil/crypto3/marshalling/algebra/processing/curve_element.hpp

```diff
@@ -1188,14 +1188,18 @@ namespace nil {
                         base_integral_type y =
                             read_data<params_type::bit_length(), base_integral_type, endianness>(iter);
                         bool sign = *(iter + encoded_size - 1) & (1 << 7);
-                        group_affine_value_type decoded_point_affine =
-                            detail::recover_x<group_affine_value_type>(y, sign);
+
+                        auto decoded_point_affine =
+                                detail::recover_x<group_affine_value_type>(y, sign);
+
+                        if (!decoded_point_affine) {
+                            return nil::marshalling::status_type::invalid_msg_data;
```

Comment:
I think here we need something like return decoded_point_affine.error()

Reply:
Fixed.